orttraining/orttraining/python/training/ort_triton/_utils.py

```diff
@@ -0,0 +1,152 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+import re
+import uuid
+from collections import defaultdict
+from typing import Any, List, Tuple
+
+import numpy as np
+from onnx import GraphProto, NodeProto, TensorProto, helper, numpy_helper
+from onnx.mapping import TENSOR_TYPE_TO_NP_TYPE
+
+
+def gen_unique_name(prefix: str) -> str:
+    return prefix + "_" + uuid.uuid4().hex[:8]
+
+
+def _topological_soft_internal(node, visited, output_consumers, sorted_nodes):
+    visited.add(node.name)
+    for next_node in output_consumers[node.name]:
```

Comment:
Python recursion limit is ~2000. Any chance that goes beyond?

Reply:
It's to process sub-graph which has only connected elementwise and reduce Ops, not likely to have such huge high.